### PR TITLE
Bind Giropay to its project

### PR DESCRIPTION
**Pull request by cmaier**
Giropay was missing the project that the user was currently check into when adding it as a payment method.
This led to weird behavior, e.g. when deleting it as a payment method.

### How to test?

* Integrate the changes from this branch into a project that's Gipay enabled.
* Add Giropay as a Payment method, check if it's binded to the project and _behaves correctly_.

### Definition of Done

- [ ] Changelog gepflegt
- [x] Dokumentation gepflegt
- [x] Alle Anforderung des Issues sind erfüllt
- [x] Selbst greviewed _(aka [Self-Reviews (eng)](https://www.nerdwallet.com/blog/engineering/why-you-should-be-doing-self-reviews/))_
- [ ] Review mit Product Owner _(Release-Variante o. Minified Build)_

#### App Tests
- [x] Minified getestet? _(aka. Release Build)_
- [x] Environements beachtet _(Production/Staging)_
- [x] Unterstützte Sprachen sind getestet
- [x] Light-/Dark-Mode getestet
- [x] Edge-Cases getestet
- [x] Android API Levels wurden berücksichtigt _(minSdk?)_

#### Testing
- ~~Tests geschrieben _(aka Unit-Tests, Integration-Tests)_~~
- [x] Tests lokal laufen gelassen

### CODE REVIEW

File: core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java [hunk 1]
```diff
@@ -364,14 +364,18 @@ private static long parseValidTo(String format, String expirationDate) {
     /**
      * Encrypts and stores a paydirekt authorization token.
      */
-    public static PaymentCredentials fromGiropay(GiropayAuthorizationData authorizationData, String customerAuthorizationURI) {
+    public static PaymentCredentials fromGiropay(
+            GiropayAuthorizationData authorizationData,
+            String customerAuthorizationURI,
+            @Nullable String projectId) {
```
Comment (Fabtron) [inline core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java:370]: If it is bound to a project shouldn't we return if no projectId is provided? In case for some reason the projectId is null, we would run into the same problem again. Or as we did for like external billing mark it with `@NonNull` to prevent it before adding it.

Comment (Fabtron) [reply core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java:370]: Especially if we're allready doing the check in the `GiropayInputView`, it would be the better way to make it not nullable i guess.

**Review by Fabtron — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java (hunk 1 above)
Comment (cmaier) [reply core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java:370]: I'll look into it

Comment (cmaier) [reply core/src/main/java/io/snabble/sdk/payment/PaymentCredentials.java]: Should fit now 👍

**Review by Fabtron — APPROVED** [review]
